File: TaskMasterVB/ThisAddIn.cs
using System.Collections.Generic;
using System.Runtime.CompilerServices;
using Microsoft.Office.Core;
using Microsoft.Office.Interop.Outlook;
using System.IO;
using ToDoModel;
using System;
using UtilitiesVB;

namespace TaskMaster
{

    public partial class ThisAddIn
    {

        private IApplicationGlobals _globals;

        public List<string> CatFilterList;
        
        private Items _OlToDoItems;
        private Items OlToDoItems
        {
            [MethodImpl(MethodImplOptions.Synchronized)]
            get
            {
                return _OlToDoItems;
            }

            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                if (_OlToDoItems != null)
                {
                    _OlToDoItems.ItemAdd -= OlToDoItems_ItemAdd;
                    _OlToDoItems.ItemChange -= OlToDoItems_ItemChange;
                }

                _OlToDoItems = value;
                if (_OlToDoItems != null)
                {
                    _OlToDoItems.ItemAdd += OlToDoItems_ItemAdd;
                    _OlToDoItems.ItemChange += OlToDoItems_ItemChange;
                }
            }
        }
        private Items _OlInboxItems;
        private Items OlInboxItems
        {
            [MethodImpl(MethodImplOptions.Synchronized)]
            get
            {
                return _OlInboxItems;
            }

            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                _OlInboxItems = value;
            }
        }
        private Reminders _OlReminders;
        private Reminders OlReminders
        {
            [MethodImpl(MethodImplOptions.Synchronized)]
            get
            {
                return _OlReminders;
            }

            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                _OlReminders = value;
            }
        }
        
        private NameSpace OlNS;

        private RibbonController _ribbonController;
        
        public readonly string FilenameDictPpl = "pplkey.xml";
        public readonly string StagingPath = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
        public string EmailRoot;
        private const string _appDataFolder = "TaskMaster";

        public ProjectInfo ProjInfo;
        public Dictionary<string, string> DictPPL;
        
        private ListOfIDs _IDList;
        public ListOfIDs IDList
        {
            [MethodImpl(MethodImplOptions.Synchronized)]
            get
            {
                return _IDList;
            }

            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                _IDList = value;
            }
        }
        
        public TreeOfToDoItems DM_CurView;
        public FlagParser Cats;

        private void ThisAddIn_Startup()
        {
            //_globals = new ApplicationGlobals(Application);

            {
                OlNS = _globals.Ol.NamespaceMAPI;
                OlToDoItems = _globals.Ol.ToDoFolder.Items;
                OlInboxItems = _globals.Ol.Inbox.Items;
                OlReminders = _globals.Ol.OlReminders;
                ProjInfo = (ProjectInfo)_globals.TD.ProjInfo;
                DictPPL = _globals.TD.DictPPL;
                IDList = (ListOfIDs)_globals.TD.IDList;
                EmailRoot = _globals.Ol.EmailRootPath;
            }

            _ribbonController.SetGlobals(_globals);
        }

        protected override IRibbonExtensibility CreateRibbonExtensibilityObject()
        {
            _ribbonController = new RibbonController();
            return new RibbonViewer(_ribbonController);
        }

        #region Explorer Event Hooks
        internal void Events_Hook()
        {
            {
                OlToDoItems = _globals.Ol.ToDoFolder.Items;
                OlInboxItems = _globals.Ol.Inbox.Items;
                OlReminders = _globals.Ol.OlReminders;
            }
        }

        internal void Events_Unhook()
        {
            OlToDoItems = null;
            OlInboxItems = null;
            OlReminders = null;
        }

        private void OlToDoItems_ItemAdd(object Item)
        {
            ToDoEvents.OlToDoItems_ItemAdd(Item, _globals);
        }

        private void OlToDoItems_ItemChange(object Item)
        {
            ToDoEvents.OlToDoItems_ItemChange(Item, OlToDoItems, _globals);
        }
        #endregion

    }
}

File: TaskMasterOld/ThisAddIn.vb
Option Explicit On
Imports System.Diagnostics
Imports System.IO
Imports Microsoft.Office.Core
Imports Microsoft.Office.Interop.Outlook
Imports Microsoft.Office.Interop
Imports Microsoft.Office.Tools
Imports Microsoft.VisualBasic.FileIO
Imports ToDoModel
Imports UtilitiesVB

Public Class ThisAddIn

    Private _globals As ApplicationGlobals

    Public CatFilterList As List(Of String)
    Private WithEvents OlToDoItems As Outlook.Items
    Private WithEvents ListOfPSTtodo As New List(Of Outlook.Items)
    Private WithEvents ListToDoItems As New List(Of Outlook.Items)
    Private WithEvents OlInboxItems As Outlook.Items
    Private WithEvents OlReminders As Outlook.Reminders
    Private OlNS As Outlook.NameSpace

    Private _ribbonController As RibbonController

    Private ReadOnly _filenameProjectList As String
    Private ReadOnly _filenameProjInfo2 As String
    Private ReadOnly _filenameProjInfo As String = "ProjInfo.bin"
    Public ReadOnly FilenameDictPpl As String = "pplkey.xml"
    Public ReadOnly StagingPath As String = SpecialDirectories.MyDocuments
    Public EmailRoot As String
    Private Const _appDataFolder = "TaskMaster"

    Public ProjInfo As ProjectInfo
    Public DictPPL As Dictionary(Of String, String)
    Public WithEvents IDList As ListOfIDs
    Public DM_CurView As TreeOfToDoItems
    Public Cats As FlagParser

    Private Sub ThisAddIn_Startup() Handles Me.Startup
        _globals = New ApplicationGlobals(Application)

        With _globals
            OlNS = .Ol.NamespaceMAPI
            OlToDoItems = .Ol.ToDoFolder.Items
            OlInboxItems = .Ol.Inbox.Items
            OlReminders = .Ol.OlReminders
            ProjInfo = .TD.ProjInfo
            DictPPL = .TD.DictPPL
            IDList = .TD.IDList
            EmailRoot = .Ol.EmailRootPath
        End With

        _ribbonController.SetGlobals(_globals)

    End Sub

    Protected Overrides Function CreateRibbonExtensibilityObject() As Microsoft.Office.Core.IRibbonExtensibility
        _ribbonController = New RibbonController()
        Return New RibbonViewer(_ribbonController)
    End Function

#Region "Explorer Event Hooks"
    Friend Sub Events_Hook()
        With _globals
            OlToDoItems = .Ol.ToDoFolder.Items
            OlInboxItems = .Ol.Inbox.Items
            OlReminders = .Ol.OlReminders
        End With
    End Sub

    Friend Sub Events_Unhook()
        OlToDoItems = Nothing
        OlInboxItems = Nothing
        OlReminders = Nothing
    End Sub

    Private Sub OlToDoItems_ItemAdd(Item As Object) Handles OlToDoItems.ItemAdd
        ToDoEvents.OlToDoItems_ItemAdd(Item, _globals)
    End Sub

    Private Sub OlToDoItems_ItemChange(Item As Object) Handles OlToDoItems.ItemChange
        ToDoEvents.OlToDoItems_ItemChange(Item, OlToDoItems, _globals)
    End Sub
#End Region

End Class
